laws/src/main/scala/cats/laws/ApplicativeErrorLaws.scala

```diff
@@ -54,6 +54,21 @@ trait ApplicativeErrorLaws[F[_], E] extends ApplicativeLaws[F] {
 
   def redeemDerivedFromAttemptMap[A, B](fa: F[A], fe: E => B, fs: A => B): IsEq[F[B]] =
     F.redeem(fa)(fe, fs) <-> F.map(F.attempt(fa))(_.fold(fe, fs))
+
+  /*
+   * These laws, taken together with applicativeErrorHandle, show that errors dominate in
+   * ap, *and* show that handle has lexical semantics over ap. F.unit is used in both laws
+   * because we don't have another way of expressing "an F[_] which does *not* contain any
```

Comment:
Another way might be `Arbitrary[F[_]].map(_.attempt)`... But that's probably problematic in other ways (depending on the behavior of attempt)

Reply:
Well, the problem is then you need a function `E => A`. That's not doable without breaking binary compatibility.

Reply:
Why binary compatibility? I fail to see that here...

Reply:
In order to have an `Arbitrary[E => A]`, we need a `Cogen[E]` and an `Arbitrary[A]`. We have the former, but not the latter (in `ApplicativeErrorTests`). In order to *get* the latter, we would need to add it as a parameter to the `applicativeError` function, which breaks binary compatibility.

Reply:
Ohhh. Yeah, you're absolutely right.